src/wallet/scriptpubkeyman.cpp

```diff
@@ -1635,4 +1635,212 @@ bool DescriptorScriptPubKeyMan::CanUpdateToWalletDescriptor(const WalletDescript
 
     return true;
 }
+
+SilentPaymentDescriptorScriptPubKeyMan::SilentPaymentDescriptorScriptPubKeyMan(WalletStorage& storage, WalletDescriptor& descriptor)
+    : DescriptorScriptPubKeyMan(storage, descriptor, 0)
+{
+    PopulateLabelTweaks();
+}
+
+void SilentPaymentDescriptorScriptPubKeyMan::PopulateLabelTweaks()
+{
+    LOCK(cs_desc_man);
+    if (m_wallet_descriptor.descriptor->GetOutputType() != OutputType::SILENT_PAYMENTS) {
+        throw std::runtime_error(std::string(__func__) + ": descriptor is not a Silent Payment Descriptor");
+    }
+    const auto provider{GetSPProvider()};
```

Comment:
It feels strange to call the provider just to get the scan key, feels like another area we can simplify with a custom scan key type.